netket/hilbert/random/base.py

```diff
@@ -12,19 +12,27 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from functools import partial, singledispatch
+from textwrap import dedent
+from typing import Tuple, Union
 
-import numpy as np
 import jax
+import numpy as np
+
+from netket.utils.dispatch import dispatch
 
+NoneType = type(None)
+Dim = Union[Tuple[int], Tuple[int, int], Tuple[int, int, int]]
 
-def random_state(hilb, key, size=None, dtype=np.float32):
+
+@dispatch
+def random_state(hilb, key, *, size=None, dtype=np.float32):
     r"""Generates either a single or a batch of uniformly distributed random states.
 
     Args:
-        size: If provided, returns a batch of configurations of the form (size, #) if size
-            is an integer or (*size, #) if it is a tuple and where # is the Hilbert space size.
-            By default, a single random configuration with shape (#,) is returned.
+        size: If provided, returns a batch of configurations of the form (size, #) if
+            size is an integer or (*size, #) if it is a tuple and where # is the Hilbert
+            space size. By default, a single random configuration with shape (#,) is
+            returned.
         out: If provided, the random quantum numbers will be inserted into this array,
              which should be of the appropriate shape (see `size`) and data type.
         rgen: The random number generator. If None, the global NetKet random
```

Comment:
is ```rgen``` still a valid argument?

Reply:
Ora a legacy argument. It's there only for backward compatibility..
I'll add a note